lmfdb/cluster_pictures/web_cluster_picture.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+from sage.all import cached_function
+
+from lmfdb import db
+from lmfdb.logger import make_logger
+wcplog = make_logger("WCP")
+
+@cached_function
+def cp_display_knowl(label, name=None, img=None):
+    if not name:
+        name = "Cluster Picture %s" % label
+    if not img:
+        img = name
+    return '<a title = "%s [clusterpicture.data]" knowl="clusterpicture.data" kwargs="label=%s">%s</a>' % (name, label, img)
+
+def cp_knowl_guts(label):
+    out = ''
+    wcp = WebClusterPicture(label)
+    if wcp.is_null():
+        return 'Cannot find cluster picture %s' % label
+    out += "<b>Cluster picture %s</b>" % label
+    out += '<div>'
+    out += '<img src="'
+    out += wcp.image()
+    out += '" alt="cluster_picture_image"/>'
+    out += '<br>Depth: '
+    out += str(wcp.depth())
+    out += '<br>Size: '
+    out += str(wcp.size())
+    out += '<br>Potential toric rank: '
+    out += str(wcp.potential_toric_rank())
+    out += '<br>Potential good reduction curve: '
+    out += str(wcp.potential_good_reduction())
+    out += '<br>Potential good reduction Jacobian: '
+    out += str(wcp.potential_good_jacobian_reduction())
+    out += '</div>'
+    #out += '<div align="right">'
+    #out += '<a href="%s">%s home page</a>' % (str(url_for("number_fields.by_label", label=label)),label)
+    #out += '</div>'
+    return out
+
+class WebClusterPicture:
+    """
+     Class for retrieving cluster picture information from the database
+    """
+    def __init__(self, label, data=None):
+        self.label = label
+        if data is None:
+            self._data = self._get_dbdata()
+        else:
```

Comment:
Writing a bunch of access methods is okay, but it might be easier to do something like
```python
if data is None:
    data = db.cluster_pictures.lookup(self.label)
self.__dict__.update(data)
```
Then you could just access the attributes directly.

Reply:
This is again done in the same way as for number fields. But if we later decide to compute some light stuff on the spot, this method would be a bit inconvenient, no?

Reply:
You can easily mix the methods, but it's really just a matter of preference.  The way you've done it is perfectly fine.